README.md

```diff
@@ -1,7 +1,158 @@
+![Cocoapods platforms](https://img.shields.io/cocoapods/p/FTAPIKit)
+![License](https://img.shields.io/cocoapods/l/FTAPIKit)
 
-## FTAPIKit templates installation
+Declarative and generic REST API framework using Codable.
+With standard implementation using URLSesssion and JSON encoder/decoder.
+Easily extensible for your asynchronous framework or networking stack.
 
-The templates can be added to Xcode simply running `make` inside `Templates/` directory.
+## Installation
+
+When using Swift package manager install using Xcode 11+
+or add following line to your dependencies:
+
+```swift
+.package(url: "https://github.com/futuredapp/FTAPIKit.git", from: "1.0.0")
+```
+
+When using CocoaPods add following line to your `Podfile`:
+
+```ruby
+pod 'FTAPIKit', '~> 1.0'
+```
+
+# Features
+
+The main feature of this library is to provide documentation-like API
+for defining web services. This is achieved using declarative
+and protocol-oriented programming in Swift.
+
+The framework provides two core protocols reflecting the physical infrastructure:
+
+- `Server` protocol defining single web service.
+- `Endpoint` protocol defining access points for resources.
+
+Combining instances of type conforming to `Server` and `Endpoint` we can build request.
+`URLServer` has convenience method for calling endpoints using `URLSession`.
+Recommended usage on application-side is implementing API client which encapsulates
+logic not provided by this framework like signing authorized endpoints
+or conforming to `URLSessionDelegate`.
+
+![Architecture](Documentation/Architecture.svg)
+
+There are many convenient `Endpoint` protocols for various use cases
+like multipart upload, automatic encoding/decoding.
+
+- `Endpoint` protocol body is empty.
+- `DataEndpoint` sends provided data in body.
+- `UploadEndpoint` sends file in body stream.
+- `MultipartEndpoint` combines body parts into body stream.
+- `RequestEndpoint` has encodable request which is encoded using server encoding.
+
+![Endpoint types](Documentation/Endpoints.svg)
+
+## Usage
+
+### Defining web service (server)
+
+Firstly we need to define our server. We recommend using structs:
+
+```swift
+struct HTTPBinServer: URLServer {
+    let baseUri = URL(string: "http://httpbin.org/")!
+    let urlSession = URLSession(configuration: .default)
+}
+```
+
+If we want to use custom formatting we just need to add our encoding/decoding configuration:
+
+```swift
+struct HTTPBinServer: URLServer {
+    ...
+
+    let decoding: Decoding = JSONDecoding { decoder in
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+    }
+    let encoding: Encoding = JSONEncoding { encoder in
+        encoder.keyEncodingStrategy = .convertToSnakeCase
+    }
+}
+```
+
+If we need to create specific request, add some headers, usually to provide
+authorization we can override default request building mechanism.
+
+```swift
+struct HTTPBinServer: URLServer {
+    var token: String?
```

Comment:
I do not like this example, because I consider OAuth as a service that requires have support of code, that is not shipped with this framework. 
I would suggest some computed property, device UUID from Device or other header like UserAgent.